### PR TITLE
fix: Use `RatingQuestion` instead of `RankingQuestion` for sentence similarity in the `ArgillaTrainer`

### LINKED ISSUE #4049: [FEATURE] RatingQuestion should be added intead of RankingQuestion
**Is your feature request related to a problem? Please describe.**

1. It looks like only LabelQuestion and RankingQuestion work when training for sentence similarity. Instead of using the RankingQuestion, it is better to use the RatingQuestion.

2. In fact, RankingQuestion does not work properly as it takes the values (which are strings) instead of the rank.

3. In addition, check that in the TrainingTask, dataset.question_by_name supports any possible name given to the LabelQuestion or RatingQuestion other than just "label".

```python
task = TrainingTask.for_sentence_similarity(
    texts=[dataset.field_by_name("premise"), dataset.field_by_name("hypothesis")],
    label=dataset.question_by_name("other_name")
)
```
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by plaguss**
# Description

Update the `ArgillaTrainer` to use `RatingQuestion` instead of `RankingQuestion` when training for `sentence-similarity` with numerical labels.

Closes #4049 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] `test/integration/client/feedback/training/test_sentence_transformers.py`

**Checklist**

- [x] I followed the style guidelines of this project
- [x] I did a self-review of my code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: tests/integration/client/feedback/training/test_sentence_transformers.py [hunk 1]
```diff
@@ -84,7 +90,14 @@ def test_prepare_for_training_sentence_transformers(
     )
     dataset.add_records(records=feedback_dataset_records * 2)
 
-    task = TrainingTask.for_sentence_similarity(formatting_func=formatting_func)
+    if formatting_func.__name__ == "formatting_func_sentence_transformers_rating_question":
+        label_strategy = RatingQuestionUnification(
+            question=RatingQuestion(name="label", values=[1, 2]), strategy="majority"
```
Comment (davidberenstein1957) [inline tests/integration/client/feedback/training/test_sentence_transformers.py]: can't we get the question form the dataset itself? `question_by_name`

**Review by davidberenstein1957 — COMMENTED** [review]
It is lookin good @plaguss, however, I think we might need to update some docs and docstrings too? Also, could you add this as a reference to the draft release highlights too?

File: tests/integration/client/feedback/training/test_sentence_transformers.py (hunk 1 above)
Comment (plaguss) [reply tests/integration/client/feedback/training/test_sentence_transformers.py]: my bad, tackled at 46737b1

**Comment by plaguss** [comment]
> It is lookin good @plaguss, however, I think we might need to update some docs and docstrings too? Also, could you add this as a reference to the draft release highlights too?

I'll do it. Now it's also solved the bug related to using `dataset.question_by_name("other_name")` when `other_name` is different from `label`

**Comment by plaguss** [comment]
@davidberenstein1957 where are we adding the references to the draft release highlights?

**Comment by davidberenstein1957** [comment]
hi @plaguss I think you might be working from a forked Argilla repo on this one?

File: docs/_source/practical_guides/fine_tune.md [hunk 2]
```diff
@@ -533,6 +533,23 @@ task = TrainingTask.for_sentence_similarity(
...
+from argilla.feedback import RatingQuestionUnification
+
+label_strategy = RatingQuestionUnification(
+    question=dataset.question_by_name("other-question"),
+    strategy="majority"
+)
+
+task = TrainingTask.for_sentence_similarity(
+    texts=[dataset.field_by_name("premise"), dataset.field_by_name("hypothesis")],
+    label=dataset.question_by_name("label"),
```
Comment (davidberenstein1957) [inline docs/_source/practical_guides/fine_tune.md]: shouldn't this be other-question?

File: docs/_source/practical_guides/fine_tune.md [hunk 3]
```diff
@@ -533,6 +533,23 @@ task = TrainingTask.for_sentence_similarity(
...
+    label_strategy=label_strategy
```
Comment (davidberenstein1957) [inline docs/_source/practical_guides/fine_tune.md]: I think we can pass "majority" directly.

File: docs/_source/practical_guides/fine_tune.md (hunk 2 above)
Comment (plaguss) [reply docs/_source/practical_guides/fine_tune.md]: yes thanks

File: docs/_source/practical_guides/fine_tune.md (hunk 3 above)
Comment (plaguss) [reply docs/_source/practical_guides/fine_tune.md]: Sure! I was working in the tests using the version with `formatting_func` which doesn't allow for that and I missed it. Should I let the example just using "majority" or passing the full class as we would do in the other case?

Comment (davidberenstein1957) [reply docs/_source/practical_guides/fine_tune.md]: If it runs by just passing "majority" we should use this simpler version IMO

**Pull request merged**